github/actions/config.go

```diff
@@ -78,16 +77,20 @@ func (c *GitHubConfig) GitHubAPIURL(path string) *url.URL {
 		Scheme: c.ConfigURL.Scheme,
 	}
 
-	switch c.ConfigURL.Host {
-	// Hosted
-	case "github.com", "github.localhost":
-		result.Host = fmt.Sprintf("api.%s", c.ConfigURL.Host)
-	// re-routing www.github.com to api.github.com
-	case "www.github.com":
-		result.Host = "api.github.com"
-
-	// Enterprise
-	default:
+	isHosted := isHostedGitHubURL(c.ConfigURL)
+
+	if isHosted {
+		// Hosted
+		if strings.EqualFold("github.com", c.ConfigURL.Host) ||
+			strings.EqualFold("github.localhost", c.ConfigURL.Host) ||
+			strings.HasSuffix(c.ConfigURL.Host, ".ghe.com") {
+			result.Host = fmt.Sprintf("api.%s", c.ConfigURL.Host)
+		} else if strings.EqualFold("www.github.com", c.ConfigURL.Host) {
+			// re-routing www.github.com to api.github.com
+			result.Host = "api.github.com"
+		}
```

Comment:
```suggestion
		if strings.EqualFold("www.github.com", c.ConfigURL.Host) {
			// re-routing www.github.com to api.github.com
			result.Host = "api.github.com"
		} else {
		        result.Host = fmt.Sprintf("api.%s", c.ConfigURL.Host)
		}
```
Since we passed the filter for hosted, can we simplify it like this?

Reply:
i like this, i even refactor further to remove all the `else`

Reply:
Your version is even better :smile: